File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());

        Queue<Integer> qe = new LinkedList<>();
        StringBuilder sb = new StringBuilder();

        for (int i = 1; i <= n; i++) {
            qe.add(i);
        }

        sb.append('<');

        while(!qe.isEmpty()) {
            for (int i = 0; i < k - 1; i++) {
                qe.add(qe.poll());
            }
            sb.append(qe.poll());

            if (!qe.isEmpty()) {
                sb.append(", ");
            }
        }

        sb.append('>');

        System.out.print(sb);
    }
}

File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.LinkedList
import java.util.StringTokenizer

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val st = StringTokenizer(readLine())
    val n = st.nextToken().toInt()
    val k = st.nextToken().toInt()
    val list = LinkedList<Int>()
    for (i in 1..n) {
        list.add(i)
    }
    print("<")
    while (list.isNotEmpty()) {
        for (i in 1..k) {
            if (i != k) {
                list.add(list.remove())
            } else if (list.size != 1) {
                print(list[0])
                print(", ")
                list.remove()
            } else {
                print(list[0])
                list.remove()
            }
        }
    }
    println(">")
}